### PR TITLE
CONSOLE-2361: Update react-catalog-view-extension to version that does not require PatternFly 3

### PR DESCRIPTION
**Pull request by rhamilto**
This PR should result in no significant changes in the rendering of the catalog, but migrate us to a version of react-catalog-view-extension that does not include Bootstrap/PatternFly 3 as a dependency. 

Notes: 
* I observed one very subtle difference in the distance between the checkbox and the checkbox label in `FilterSidePanelCategoryItem`, but the difference is negligible.

### CODE REVIEW

**Comment by rhamilto** [comment]
/hold for approvals
/assign @yapei

@yapei We just need a regression test to make sure there are no visual problems updating this dependency.

**Review by dtaylor113 — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
/hold

See https://github.com/patternfly/patternfly/issues/4155#issuecomment-877412085

I think we should wait for the revert, and then the override I added won't be necessary.

**Comment by yanpzhan** [comment]
Tested the pr, did some regression test on operatorhub page and catalog page. Didn't find issue. 
@rhamilto  if QE needs test again after patternfly issue reverts and your fix for it is removed?

**Comment by rhamilto** [comment]
> Tested the pr, did some regression test on operatorhub page and catalog page. Didn't find issue.
> @rhamilto if QE needs test again after patternfly issue reverts and your fix for it is removed?

@yanpzhan, I don't think a second QE test is necessary since the gist of this PR is to migrate to a version of react-catalog-view-extension that does not include Bootstrap/PatternFly 3 as a dependency, which will not require retesting.

**Comment by rhamilto** [comment]
I found and removed more [orphaned CSS](https://github.com/openshift/console/pull/9419/files#diff-28c7abc9fceeb716adb6850a96da67b203a13428b8ca95649c3ae66834b3573bL52-L74).  Please retag.

File: frontend/public/style/_overrides.scss [hunk 1]
```diff
@@ -1,5 +1,4 @@
 // Use this file to override styles from 3rd party dependencies
-$pf-4-nav-bar-height: 76px; // Height of the PatternFly 4 masthead
```
Comment (rhamilto) [inline frontend/public/style/_overrides.scss:2]: This really should have gone with https://github.com/openshift/console/pull/9452 as it is orphaned, but @spadgett already tagged that and I don't want him to have to retag it.

**Comment by rhamilto** [comment]
/hold cancel

The PatternFly regression was reverted, and this PR now makes use of the PatternFly version without the regression.

**Comment by spadgett** [comment]
It looks like the catalog extension is getting pulled into the main vendor bundle causing the analyze job failure.

**Comment by spadgett** [comment]
Or maybe just Patternfly is a little bigger, and we went over the max.

**Comment by rhamilto** [comment]
> Or maybe just Patternfly is a little bigger, and we went over the max.

I'm thinking this must be the case as I can't see a change that would result a difference.

**Comment by spadgett** [comment]
See https://github.com/openshift/console/pull/9069#discussion_r669861607. We added a new library, but didn't increase the vendor bundle limit enough IMO.

**Comment by spadgett** [comment]
Let's increase the limit to something like 3.25MB for now and we can revisit whether we want to lazy load the quick starts library separately.

cc @jschuler

**Comment by rhamilto** [comment]
> Let's increase the limit to something like 3.25MB for now and we can revisit whether we want to lazy load the quick starts library separately.

[Done](https://github.com/openshift/console/pull/9419/files#diff-1141b1b0eb99550ca4c4e84b35820a2cd4a56da52bf31f6febd6b75dd15416d6L17).

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**